Dockerfile

```diff
@@ -1,56 +1,38 @@
-# Use a specific Python 3.11 slim image as the base
-FROM python:3.11.3-slim
-
-# Set environment variables
-ENV PYTHONDONTWRITEBYTECODE=1 \
-    PYTHONUNBUFFERED=1
-
-# Install system dependencies
-RUN apt-get update && apt-get install -y --no-install-recommends \
-    git \
-    curl \
-    unzip \
-    xvfb \
-    libnss3 \
-    libatk-bridge2.0-0 \
-    libgtk-3-0 \
-    libxss1 \
-    libasound2 \
-    libx11-xcb1 \
-    libxcb-dri3-0 \
-    libdrm2 \
-    libgbm1 \
-    libxcomposite1 \
-    libxdamage1 \
-    libxrandr2 \
-    libxinerama1 \
-    libxcursor1 \
-    libxi6 \
-    libgl1-mesa-glx \
-    && rm -rf /var/lib/apt/lists/*
-
-# Create a non-root user
-RUN addgroup --system appgroup && adduser --system --ingroup appgroup appuser
-
-# Set work directory
+# 1. Base image
+FROM python:3.11-slim
+
+# 2. Create a non-root user (don’t switch yet)
+RUN groupadd appgroup \
+ && useradd -m -g appgroup appuser
+
+# 3. Install base APT packages
+RUN apt-get update -qq \
+ && apt-get install -y --no-install-recommends -qq \
+      unzip libnss3 libxss1 libasound2 libx11-xcb1 \
+      curl git \
+ && rm -rf /var/lib/apt/lists/*
+
+# 4. Set working directory
 WORKDIR /app
 
-# Copy project files
-COPY . /app/
+# 5. Copy only dependency manifest
+COPY pyproject.toml /app/
```

Comment:
this is not used anywhere currently, you need to also extract the playwright version somehow or tell pip to use it somehow